expo/expo-cli
# [webpack] Prevent installing sharp in expo-cli packages

## Why 

- fix #1301
- `sharp` fails to install a lot.
- `sharp` adds a lot of extra install time to `expo-cli, @expo/webpack-config, expo-optimize`, basically most packages.

## How

- Prompt users to install `sharp` themselves
- If sharp isn't installed, use `jimp` to process images
- Bail out gracefully from `expo-optimize` if sharp isn't installed manually 
- Document that `sharp-cli` should be installed in the `expo-optimize` docs

- In webpack we cache all of the generated images in `.expo` so rebuild times won't be too effected most of the time. #1262
- In cases where all images could be restored from the cache nothing will print out
- When images need to be generated and sharp isn't installed, the user will get a prompt about how to install sharp.

## Test Plan

- try building web with sharp installed globally and locally - should install fast like normal
- try building web with sharp uninstalled globally and locally - images should generate slower but still look the same as the sharp images
- rebuild and yarn updates should be much faster too.
